text/0000-fabric-x-next-generation.md

```diff
@@ -0,0 +1,365 @@
+* **Decoupled Endorser:** By extracting the endorser component, Fabric-X allows the endorsing logic to be embedded directly within applications or run as a closely coupled service, using an **endorser SDK**. This SDK provides the tools to construct the read-write set and generate the endorsement signature according to the application's business logic.
+* **Flow-Based Programming with FSC:** Instead of deploying chaincode (shim interface) to peers, developers will typically design and implement 'Flows' using the Fabric Smart Client. Flows are sophisticated orchestrations that can involve:
+    * Complex off-chain interactions and negotiations between parties.
+    * Integration with external systems or other blockchains.
+    * Invocation of on-chain logic (via the endorser SDK to build transaction proposals).
+    * Diverse endorsement logic run by different organizations for the same transaction, catering to complex regulatory or business requirements.
+* **Namespaces and Endorsement Policies:** As in HLF, transactions are associated with namespaces, each protected by an endorsement policy (defined using Fabric's policy DSL) that specifies the required set of endorsing parties.
+This enhanced programming model allows developers to build more powerful, interactive, and integrated distributed applications while still benefiting from Fabric's core security model based on endorsement policies. While this represents a shift from traditional chaincode, it aims to improve developer productivity and address more complex use cases. Support for running existing Fabric chaincode directly is not an initial goal, as the execution context and APIs would differ.
+
+## 3. Threshold-Based Signatures and Optimized Traditional Policies
+
+To address limitations 4 (Large Transaction Size) and 5 (Endorsement Policy Verification Overhead):
+* Fabric-X allows endorsement policies to be defined by a single public key, thereby requiring only one signature. For multi-org endorsed transactions, Fabric-X will introduce support for threshold signature schemes, accompanied by a key distribution framework. This approach enables a transaction to carry a single, compact threshold signature instead of multiple individual signatures and certificates. Consequently, this significantly reduces both transaction size and the computational cost of verification. The verification key for the threshold signature can be directly linked to the endorsement policy.
+* To retain expressiveness, Fabric-X will add support for traditional endorsement policies (e.g., AND/OR/OutOf). To mitigate their impact on transaction size, identities (such as X.509 certificate information) can be pre-registered on the blockchain, for example, via configuration blocks or the "known identities" feature within the MSP folder. Transactions can then reference these known, pre-registered identities along with the signature, eliminating the need to embed full certificates in every transaction.
```

Comment:
> To retain expressiveness, Fabric-X will add support for traditional endorsement policies (e.g., AND/OR/OutOf)

Ok, so this is not something you have already implemented, and this is a future intent?

Reply:
This is currently in development.